impl-tinkerpop/src/test/java/org/hawkular/inventory/impl/tinkerpop/test/BasicTest.java

```diff
@@ -283,8 +295,12 @@ public void testRelationshipServiceGet() throws Exception {
             "going from  environments must have uid equal to 'test'. Was: " + contains.getTarget().getId();
         assert "contains" .equals(contains.getName()) : "Name of the relation must be 'contains'.";
 
-        contains = inventory.tenants().getAll().environments().get("test").relationships().get("13").entity();
-        assert contains == null : "There should not be any edge with id 13.";
+        try {
+            inventory.tenants().getAll().environments().get("test").relationships().get("13").entity();
```

Comment:
+1, I also think we should really move the testsuite to the API and make it "abstract" in that it would need to be "seeded" with a concrete implementation.

The tinkerpop impl would then just instantiate the testsuite, seed it with the tinkerpop inventory impl and run it.

This way we define the required behavior generically and impls will have something to conform to.

Blueprints testsuite is written this way btw.